File: src/main/java/com/morph/engine/graphics/FontResource.java
package com.morph.engine.graphics;

public class FontResource {
	private LoadedFont font;
	private int count;

	public FontResource(LoadedFont font) {
		this.font = font;
	}
	
	@Override
	public void finalize() {
		font.destroy();
	}
	
	public LoadedFont getFont() {
		return font;
	}
	
	public void addReference() {
		count++;
	}
	
	public boolean removeReference() {
		count--;
		return count == 0;
	}
}

File: src/main/kotlin/com/morph/engine/graphics/FontResource.kt
package com.morph.engine.graphics

class FontResource(val font: LoadedFont) {
    private var count: Int = 0

    fun finalize() {
        font.destroy()
    }

    fun addReference() {
        count++
    }

    fun removeReference(): Boolean {
        count--
        return count == 0
    }
}
